### ISSUE TITLE
`#[inline_props]` + `#[props(!optional)]` doesn't work with `pub fn`

### ISSUE THREAD

**Issue by nekomatamune** [DioxusLabs/dioxus#583]
**Problem**

The following function does not compile
```rs
#[inline_props]
pub fn view4(cx: Scope, #[props(!optional)] _v: Option<u32>) -> Element {
    cx.render(rsx!{ "view4" })
}
```

Error message:
```sh
error: expected identifier, found `#`
  --> src/main.rs:36:25
   |
36 | pub fn view4(cx: Scope, #[props(!optional)] _v: Option<u32>) -> Element {
   |                         ^ expected identifier

error: expected identifier
  --> src/main.rs:36:25
   |
36 | pub fn view4(cx: Scope, #[props(!optional)] _v: Option<u32>) -> Element {
   |                         ^

```

**Steps To Reproduce**

- Clone and build [this repo](https://github.com/nekomatamune/dioxus-bug/tree/issue-583)

**Expected behavior**

Expect that `#[inline_prop]` + `#[props(!optional)` should work on public function. 

In fact, any of the following combinations does work
- `#[inline_prop]` + `#[props(!optional)` 
- `#[inline_prop]` +  `pub fn`
- `#[props(!optional)` + `pub fn`

But not when all three are used together
- `#[inline_prop]` + `#[props(!optional)` + `pub fn`

**Environment:**
 - Dioxus version: 0.2.4
 - Rust version: 1.61.0
 - OS info: MacOS
 - App platform: n/a

**Comment by DogeDark** [comment]
I looked into this, and the compiler error seems to come from `core-macro/src/lib.rs` line 22. The `parse_macro_input` macro from the `syn` crate:

```rs
#[proc_macro_derive(Props, attributes(props))]
pub fn derive_typed_builder(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
    let input = parse_macro_input!(input as syn::DeriveInput); // Error
    match props::impl_my_derive(&input) {
        Ok(output) => output.into(),
        Err(error) => error.to_compile_error().into(),
    }
}
```
This is my first time working with macros, and I'm unsure how to help further. I suspect this issue might be on syn's side. I'll leave this here for when someone with more experience looks into it.

**Tested on:**
- Dioxus 0.2.4 & current master branch
- Rust version: 1.64.0
- OS: Windows 10

**Comment by tigerros** [comment]
Fixed in #1448
